models/notifications/requests/request.py

```diff
@@ -0,0 +1,52 @@
+class Request(object):
+    """ Base class used for requests to represent a notification payload.
+
+    Attributes:
+        notification (Notification): The Notification to send.
+    """
+
+    def __init__(self, notification):
+        """
+        Args:
+            notification (Notification): The Notification to send.
+        """
+        self.notification = notification
+
+    def send(self):
+        """ NotificationRequests should understand how to send themselves to wherever they are going.
+
+        Returns:
+            NotificationResponse
+        """
+        raise NotImplementedError("NotificationRequest subclass must implement send")
+        # deferred.defer(self.track_notification, self._type, num_keys, _queue="api-track-call", _url='/_ah/queue/deferred_notification_track_send')
+
+    def track_notification(self, notification_type_enum, num_keys):
+        """
+        For more information about GAnalytics Protocol Parameters, visit
+        https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters
+        """
+        import urllib
+        analytics_id = Sitevar.get_by_id("google_analytics.id")
```

Comment:
Would love to swap out this code with https://github.com/the-blue-alliance/the-blue-alliance/pull/2582